src/solvers/eigsolve.jl

```diff
@@ -0,0 +1,40 @@
+
+function eigsolve_solver(;
+    solver_which_eigenvalue=:SR,   #TODO: settle on pattern to pass solver kwargs
+    ishermitian=true,
+    solver_tol=1e-14,
+    solver_krylovdim=3,
+    solver_maxiter=1,
+    solver_verbosity=0,
+  )
+    
+    function solver(
+        init;
+        psi_ref!,
+        PH_ref!,
```

Comment:
I don't think it's obvious but I would vote for just calling these `psi!` and `PH!` (and ideally more descriptive names like `state!` and `projected_operator!`). We can just have a convention that keyword arguments ending in `!` are `Ref`s.

Reply:
I am ok with that. I may keep it around for a while to be more explicit, but will ultimately rename them.